### PR TITLE
Fetch `Calling/src` from upstream

### PR DESCRIPTION
**Pull request by prprabhu-ms**
Fetched from upstream at [2eb5ba](https://github.com/Azure/communication-ui-library/tree/2eb5ba8713e2fcfc9eca9c2328b56d9f42cf1361/samples/Calling)

# How Verified

Locally ran an 8 person call

### CODE REVIEW

File: Calling/src/app/App.tsx [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
+import { GroupCallLocator, TeamsMeetingLinkLocator } from '@azure/communication-calling';
 import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { setLogLevel } from '@azure/logger';
 import { initializeIcons, Spinner } from '@fluentui/react';
+import { CallAdapterLocator } from '@internal/react-composites';
 import React, { useEffect, useState } from 'react';
 import {
   buildTime,
   callingSDKVersion,
+  communicationReactSDKVersion,
```
Comment (prprabhu-ms) [inline Calling/src/app/App.tsx:13]: Delete this upstream-ism

File: Calling/src/app/App.tsx [hunk 2]
```diff
@@ -1,38 +1,44 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-import { GroupCallLocator, GroupLocator, TeamsMeetingLinkLocator } from '@azure/communication-calling';
+import { GroupCallLocator, TeamsMeetingLinkLocator } from '@azure/communication-calling';
 import { CommunicationUserIdentifier } from '@azure/communication-common';
+import { setLogLevel } from '@azure/logger';
 import { initializeIcons, Spinner } from '@fluentui/react';
+import { CallAdapterLocator } from '@internal/react-composites';
```
Comment (prprabhu-ms) [inline Calling/src/app/App.tsx:8]: Use @azure/communication-react

File: Calling/src/app/App.tsx [hunk 3]
```diff
@@ -59,35 +65,53 @@ const App = (): JSX.Element => {
...
             setDisplayName(callDetails.displayName);
+
             const isTeamsCall = !!callDetails.teamsLink;
-            const callLocator =
-              callDetails.teamsLink || getTeamsLinkFromUrl() || getGroupIdFromUrl() || createGroupId();
-            setCallLocator(callLocator);
+            const makeLocator = (
+              teamsLink?: TeamsMeetingLinkLocator | undefined,
+
+              outboundParticipants?: string[]
```
Comment (prprabhu-ms) [inline Calling/src/app/App.tsx:102]: outboundParticipants is fishy. Is this related to the still in beta PSTN calling feature?

File: Calling/src/app/theming/SwitchableFluentThemeProvider.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@
 // Licensed under the MIT license.
 
 import React, { useState, useMemo, createContext, useContext } from 'react';
-import { FluentThemeProvider, lightTheme, darkTheme } from '@azure/communication-react';
+import { FluentThemeProvider, lightTheme, darkTheme } from '@internal/react-components';
```
Comment (prprabhu-ms) [inline Calling/src/app/theming/SwitchableFluentThemeProvider.tsx:5]: Use `@azure/communication-react`

File: Calling/src/app/utils/AppUtils.ts [hunk 5]
```diff
@@ -57,17 +57,19 @@ export const isOnIphoneAndNotSafari = (): boolean => {
...
-  !!window.navigator.userAgent.match(/(iPad|iPhone|iPod|Android|webOS|BlackBerry|Windows Phone)/g);
-
-export const isSmallScreen = (): boolean => window.innerWidth < 700 || window.innerHeight < 400;
+export const isLandscape = (): boolean => window.innerWidth < window.innerHeight;
 
 export const navigateToHomePage = (): void => {
   window.location.href = window.location.href.split('?')[0];
 };
 
+export const WEB_APP_TITLE = document.title;
```
Comment (prprabhu-ms) [inline Calling/src/app/utils/AppUtils.ts:66]: Drop all webpack injected globals

File: Calling/src/app/utils/useIsMobile.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { useEffect, useState } from 'react';
+import MobileDetect from 'mobile-detect';
```
Comment (prprabhu-ms) [inline Calling/src/app/utils/useIsMobile.ts:5]: Note: make sure `mobile-detect` depenedency is added to package.json

File: Calling/src/app/views/CallScreen.tsx [hunk 7]
```diff
@@ -1,62 +1,38 @@
...
-
-const detectMobileSession = (): boolean => !!new MobileDetect(window.navigator.userAgent).mobile();
+import { WEB_APP_TITLE } from '../utils/AppUtils';
+import { useIsMobile } from '../utils/useIsMobile';
 
 export interface CallScreenProps {
   token: string;
   userId: CommunicationUserIdentifier;
-  callLocator: GroupCallLocator | TeamsMeetingLinkLocator;
+  callLocator: CallAdapterLocator;
```
Comment (prprabhu-ms) [inline Calling/src/app/views/CallScreen.tsx:23]: Double check that `CallAdapterLocator` has shipped to `stable` build.

File: Calling/src/app/views/CallScreen.tsx [hunk 8]
```diff
@@ -1,62 +1,38 @@
...
   displayName: string;
-  webAppTitle: string;
+  alternateCallerId?: string;
```
Comment (prprabhu-ms) [inline Calling/src/app/views/CallScreen.tsx:25]: `aleternateCallerId` is fishy. Is this related to the beta-only PSTN calling feature?

File: Calling/src/app/views/HomeScreen.tsx [hunk 9]
```diff
@@ -82,6 +90,10 @@ export const HomeScreen = (props: HomeScreenProps): JSX.Element => {
                 onChange={(_, newValue) => newValue && setTeamsLink({ meetingLink: newValue })}
               />
             )}
+
```
Comment (prprabhu-ms) [inline Calling/src/app/views/HomeScreen.tsx:93]: Remove these conditional compilation artifacts.

File: Calling/src/app/views/HomeScreen.tsx [hunk 10]
```diff
@@ -45,9 +46,15 @@ export const HomeScreen = (props: HomeScreenProps): JSX.Element => {
   const [chosenCallOption, setChosenCallOption] = useState<IChoiceGroupOption>(callOptions[0]);
   const [teamsLink, setTeamsLink] = useState<TeamsMeetingLinkLocator>();
 
+  const startGroupCall: boolean = chosenCallOption.key === 'ACSCall';
   const teamsCallChosen: boolean = chosenCallOption.key === 'TeamsMeeting';
-  const buttonEnabled = displayName && (!teamsCallChosen || teamsLink);
 
+  const buttonEnabled = displayName && (startGroupCall || teamsLink || (teamsCallChosen && teamsLink));
+
+  /**
```
Comment (prprabhu-ms) [inline Calling/src/app/views/HomeScreen.tsx:54]: This comment and `buttonEnabled` should be conditionally compiled.

**Review by JamesBurnside — APPROVED** [review]
